test/modm/mock/iodevice.hpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2009-2012, Fabian Greif
+ * Copyright (c) 2020, Sascha Schade
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_TEST_MOCK_IODEVICE_HPP
+#define MODM_TEST_MOCK_IODEVICE_HPP
+
+#include <modm/io/iodevice.hpp>
+#include <cstring>  // memset
+
+namespace modm_test::platform
+{
+
+// ----------------------------------------------------------------------------
+// simple IODevice which stores all data in a memory buffer
+// used for testing the output of an IOStream
+
+class IODevice : public modm::IODevice
+{
+public:
+	IODevice() :
+		bytesWritten(0) {}
+
+	/// Write a single char to the buffer.
+	virtual void
+	write(char c)
+	{
+		this->buffer[this->bytesWritten] = c;
+		this->bytesWritten++;
+	}
+
+	using modm::IODevice::write;
+
+	virtual void
+	flush()
+	{
+		memset(this->buffer, 0, this->buffer_length);
```

Comment:
This could call `clear()` to avoid code duplication. Also, in my opinion it would be idiomatic C++ to use std::array instead of a C-array and memset.